src/plugins/map/MQ2MapCommands.cpp

```diff
@@ -1364,6 +1580,387 @@ static void DrawMapSettings_Colors()
+			gOverrideMapLocParams.color.Alpha = 255;
+		}
+
+		gDefaultMapLocParams.color.Blue = static_cast<uint8_t>(mapLocColor.Value.z * 255);
+		gDefaultMapLocParams.color.Green = static_cast<uint8_t>(mapLocColor.Value.y * 255);
+		gDefaultMapLocParams.color.Red = static_cast<uint8_t>(mapLocColor.Value.x * 255);
+		gDefaultMapLocParams.color.Alpha = 255;
+
+		WritePrivateProfileInt("MapLoc", "Red", gDefaultMapLocParams.color.Red, INIFileName);
+		WritePrivateProfileInt("MapLoc", "Green", gDefaultMapLocParams.color.Green, INIFileName);
+		WritePrivateProfileInt("MapLoc", "Blue", gDefaultMapLocParams.color.Blue, INIFileName);
+		UpdateDefaultMapLocInstances();
+		regenerate = true;
+	}
+
+	ImGui::Separator();
+
+	ImColor radiusColor = gDefaultMapLocParams.circleColor.ToImColor();
+	if (ImGui::ColorEdit3("MapLoc Radius", &radiusColor.Value.x))
+	{
+		if (!IsMapLocsStyleOverridden())
+		{
+			gOverrideMapLocParams.circleColor.Blue = static_cast<uint8_t>(radiusColor.Value.z * 255);
+			gOverrideMapLocParams.circleColor.Green = static_cast<uint8_t>(radiusColor.Value.y * 255);
+			gOverrideMapLocParams.circleColor.Red = static_cast<uint8_t>(radiusColor.Value.x * 255);
+			gOverrideMapLocParams.circleColor.Alpha = 255;
+		}
+
+		gDefaultMapLocParams.circleColor.Blue = static_cast<uint8_t>(radiusColor.Value.z * 255);
+		gDefaultMapLocParams.circleColor.Green = static_cast<uint8_t>(radiusColor.Value.y * 255);
+		gDefaultMapLocParams.circleColor.Red = static_cast<uint8_t>(radiusColor.Value.x * 255);
+		gDefaultMapLocParams.circleColor.Alpha = 255;
+
+		WritePrivateProfileInt("MapLoc", "RadiusRed", gDefaultMapLocParams.circleColor.Red, INIFileName);
+		WritePrivateProfileInt("MapLoc", "RadiusGreen", gDefaultMapLocParams.circleColor.Green, INIFileName);
+		WritePrivateProfileInt("MapLoc", "RadiusBlue", gDefaultMapLocParams.circleColor.Blue, INIFileName);
+		UpdateDefaultMapLocInstances();
+		regenerate = true;
+	}
+
+	ImGui::Separator();
+
+	if (regenerate)
+	{
+		MapClear();
+		MapGenerate();
+	}
+}
+
+static bool IsAnyMapLocSelected()
+{
+	for (size_t i = 0; i < gMapLocTemplates.size(); i++)
+	{
+		if (gMapLocTemplates[i]->GetMapLoc()->m_isSelected)
+		{
+			return true;
+		}
+	}
+
+	return false;
+}
+
+static void DeleteSelectedMapLocs()
+{
+	for (size_t i = 0; i < gMapLocTemplates.size(); i++)
+	{
+		if (gMapLocTemplates[i]->GetMapLoc()->m_isSelected)
+		{
+			DeleteMapLoc(gMapLocTemplates[i--]->GetMapLoc());
+		}
+	} 
+}
+
+static void ResetSelectedMapLocsToDefault()
+{
+	for (size_t i = 0; i < gMapLocTemplates.size(); i++)
+	{
+		if (gMapLocTemplates[i]->GetMapLoc()->m_isSelected)
+		{
+			MapObjectMapLoc* thisMapLoc = gMapLocTemplates[i]->GetMapLoc();
+			gMapLocTemplates[i]->SetCreatedFromDefaults(true);
+			gMapLocTemplates[i]->UpdateFromParams(gDefaultMapLocParams);
+		}
+	}
+}
+
+static void ApplyOverridesToSelected(MapLocParams params)
+{
+	for (size_t i = 0; i < gMapLocTemplates.size(); i++)
```

Comment:
use ranged-based for loop, you don't need to keep indexing gMapLocTemplates (same with the function above htis one)